programs/fileio.c

```diff
@@ -1754,8 +1822,26 @@ int FIO_decompressMultipleFilenames(const char** srcNamesTable, unsigned nbFiles
                     && strcmp(suffixPtr, ZSTD_EXTENSION)
                     && strcmp(suffixPtr, LZMA_EXTENSION)
                     && strcmp(suffixPtr, LZ4_EXTENSION)) ) {
-                DISPLAYLEVEL(1, "zstd: %s: unknown suffix (%s/%s/%s/%s/%s expected) -- ignored \n",
-                             srcFileName, GZ_EXTENSION, XZ_EXTENSION, ZSTD_EXTENSION, LZMA_EXTENSION, LZ4_EXTENSION);
+                DISPLAYLEVEL(1, "zstd: %s: unknown suffix ("
+                                ZSTD_EXTENSION
+#ifdef ZSTD_GZCOMPRESS
```

Comment:
Looks like Visual doesn't like macro definitions inside macro invocations.
We can refactor to something like:

```c
const char[] kSupportedExtensions = ZSTD_EXTENSION
#ifdef ZSTD_GZCOMPRESS
  /* ... */
#endif
;
DISPLAYLEVEL(1, "zstd: %s: unknown suffix (%s expected) -- ignored \n", srcFileName, kkSupportedExtensions);
```

Reply:
So doing that (using a const built string) will make GCC unhappy (see 326224f)... I'm thinking that it might be time to break out some [string concatenation](https://gcc.gnu.org/onlinedocs/cpp/Concatenation.html) or to alternatively `ifdef` a special case for Visual.

Also, if compiler compatibility is so paramount, might it be worthwhile to consider adding jobs on Travis / Circle to check Clang too?